src/main/java/net/minecraftforge/common/PlantType.java

```diff
@@ -41,24 +43,33 @@
 
     /**
      * Getting a custom {@link PlantType}, or an existing one if it has the same name as that one. Your plant should implement {@link IPlantable}
-     * and return this custom type in {@link IPlantable#getPlantType(IBlockAccess, BlockPos)}.
+     * and return this custom type in {@link IPlantable#getPlantType(IBlockReader, BlockPos)}.
      *
      * <p>If your new plant grows on blocks like any one of them above, never create a new {@link PlantType}.
-     * This enumeration is only functioning in
-     * {@link net.minecraft.block.Block#canSustainPlant(IBlockState, IWorldReader, BlockPos, EnumFacing, IPlantable)},
+     * This Type is only functioning in
+     * {@link net.minecraft.block.Block#canSustainPlant(BlockState, IBlockReader, BlockPos, Direction, IPlantable)},
      * which you are supposed to override this function in your new block and create a new plant type to grow on that block.
      *
-     * <p>You can create an instance of your plant type in your API and let your/others mods access it. It will be faster than calling this method.
+     * This method can be called during parallel loading
      * @param name the name of the type of plant, you had better follow the style above
      * @return the acquired {@link PlantType}, a new one if not found.
      */
     public static PlantType get(String name)
     {
-        return VALUES.computeIfAbsent(name, e -> {
-            if (INVALID_CHARACTERS.matcher(e).find())
-                throw new IllegalArgumentException("PlantType.get() called with invalid name: " + name);
-            return new PlantType(e);
-        });
+        PlantType type = VALUES.get(name);
```

Comment:
I don't think calling get() while another thread is in the middle of computeIfAbsent is thread-safe

Reply:
So I did some reading, and it turns out computeIfAbset that ConcurrentHashMap actually allows the behaviour as I used it, however, computeIfAbsent is already enoguh, as the implementation for ConcurrentHashMap is written so it guranteeres that not race condition will occur